napari/layers/tracks/_track_utils.py

```diff
@@ -0,0 +1,360 @@
+
+        # check that graph nodes are of correct format
+        for node_idx, parents_idx in graph.items():
+            # make sure parents are always a list
+            if type(parents_idx) != list:
+                graph[node_idx] = [parents_idx]
+
+        # check that graph nodes exist in the track id lookup
+        for node_idx, parents_idx in graph.items():
+            nodes = [node_idx] + graph[node_idx]
+            for node in nodes:
+                if node not in self.unique_track_ids:
+                    raise ValueError(f'graph node {node_idx} not found')
+
+        return graph
+
+    def build_tracks(self):
+        """ build the tracks """
+
+        points_id = []
+        track_vertices = []
+        track_connex = []
+
+        # NOTE(arl): this takes some time when the number of tracks is large
+        for idx in self.unique_track_ids:
+            indices = self._vertex_indices_from_id(idx)
+
+            # grab the correct vertices and sort by time
+            vertices = self.data[indices, 1:]
+            vertices = vertices[vertices[:, 0].argsort()]
+
+            # coordinates of the text identifiers, vertices and connections
+            points_id += [idx] * vertices.shape[0]
+            track_vertices.append(vertices)
+            track_connex.append(connex(vertices))
+
+        self._points_id = np.array(points_id)[self._ordered_points_idx]
+        self._track_vertices = np.concatenate(track_vertices, axis=0)
+        self._track_connex = np.concatenate(track_connex, axis=0)
+        # print(f'updated data: {self._track_vertices.shape}')
+
+    def build_graph(self):
+        """ build the track graph """
+
+        graph_vertices = []
+        graph_connex = []
+
+        for node_idx, parents_idx in self.graph.items():
+            # we join from the first observation of the node, to the last
+            # observation of the parent
+            node_start = self._vertex_indices_from_id(node_idx)[0]
+            node = self.data[node_start, 1:]
+
+            for parent_idx in parents_idx:
+                parent_stop = self._vertex_indices_from_id(parent_idx)[-1]
+                parent = self.data[parent_stop, 1:]
+
+                verts = np.stack([node, parent], axis=0)
+
+                graph_vertices.append(verts)
+                graph_connex.append([True, False])
+
+        # if there is a graph, store the vertices and connection arrays,
+        # otherwise, clear the vertex arrays
+        if graph_vertices:
+            self._graph_vertices = np.concatenate(graph_vertices, axis=0)
+            self._graph_connex = np.concatenate(graph_connex, axis=0)
+        else:
+            self._graph_vertices = None
+            self._graph_connex = None
+
+    def vertex_properties(self, color_by: str) -> np.ndarray:
+        """ return the properties of tracks by vertex """
+
+        if color_by not in self.properties:
+            raise ValueError(f'Property {color_by} type not found')
```

Comment:
```suggestion
            raise ValueError(f'Property {color_by} not found')
```
? The "type" confuses me...